examples/04-signals.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$loop = React\EventLoop\Factory::create();
+
+$loop->addSignal(SIGINT, $func = function ($signal) use ($loop, &$func) {
+    echo 'Signal: ', (string)$signal, PHP_EOL;
+    $loop->removeSignal(SIGINT, $func);
+    $loop->stop();
+});
+
+// This timer is here to keep the loop alive, signaling by itself doesn't do that
```

Comment:
This appears unexpected to me and is apparently undocumented. I'm curious for the reasoning and documentation here, as I would clearly have expected this to behave otherwise. What is the motivation for not making this "keep the loop alive"? (Keeping consistency with other event loop listeners in mind here)

Also note that this is related to https://github.com/reactphp/event-loop/issues/107.

Reply:
The motivation to not keeping the loop alive with signals was that there where not inputs into the loop kicking off more activity besides the signals. But now that I think about it signals COULD be used to kick off more activity. I'll update this PR shortly using a timer to keep the loop alive as long as signals are registered to it.